### PR TITLE
Update dashboard metrics to match new specification

### LINKED ISSUE #103: Update metrics dashboard based on new definitions
See changes from https://github.com/software-platform/monorepo/issues/102 and implement changes in the metrics project.

### PR DESCRIPTION
**Pull request by vova-beloded**
Fixes #103 

## Checklist

- [x] Documented
- [x] Unit tested
- [x] Component tested
- [x] Driver tested

### CODE REVIEW

File: metrics/lib/features/dashboard/domain/usecases/receive_project_metrics_updates.dart [hunk 1]
```diff
@@ -18,10 +19,8 @@ import 'package:rxdart/rxdart.dart';
 /// Provides an ability to get the [DashboardProjectMetrics] updates.
 class ReceiveProjectMetricsUpdates
     implements UseCase<Stream<DashboardProjectMetrics>, ProjectIdParam> {
-  static const int numberOfBuildResults = 14;
-  static const Duration buildMetricsLoadingPeriod = Duration(days: 7);
-  static const Duration averageBuildDurationCalculationPeriod =
-      Duration(days: 14);
+  static const int numberOfLastBuilds = 14;
+  static const Duration buildsLoadingPeriod = Duration(days: 7);
```
Comment (illia-romanenko) [inline metrics/lib/features/dashboard/domain/usecases/receive_project_metrics_updates.dart]:
```suggestion
  static const Duration buildNumberLoadingPeriod = Duration(days: 7);
```

File: metrics/lib/features/dashboard/domain/usecases/receive_project_metrics_updates.dart [hunk 2]
```diff
@@ -18,10 +19,8 @@ import 'package:rxdart/rxdart.dart';
 /// Provides an ability to get the [DashboardProjectMetrics] updates.
 class ReceiveProjectMetricsUpdates
     implements UseCase<Stream<DashboardProjectMetrics>, ProjectIdParam> {
-  static const int numberOfBuildResults = 14;
-  static const Duration buildMetricsLoadingPeriod = Duration(days: 7);
-  static const Duration averageBuildDurationCalculationPeriod =
-      Duration(days: 14);
+  static const int numberOfLastBuilds = 14;
```
Comment (illia-romanenko) [inline metrics/lib/features/dashboard/domain/usecases/receive_project_metrics_updates.dart]:
```suggestion
  static const int lastBuildsForChartsMetrics = 14;
```

File: metrics/lib/features/dashboard/presentation/widgets/titled_text.dart [hunk 3]
```diff
@@ -0,0 +1,62 @@
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:metrics/features/dashboard/presentation/widgets/expandable_text.dart';
+
+/// The widget stands for displaying the [value] text with some [title].
+class TitledText extends StatelessWidget {
```
Comment (illia-romanenko) [inline metrics/lib/features/dashboard/presentation/widgets/titled_text.dart]: Maybe - ValueBox? Or TextMetric? I have a feeling that there should be a better name for it.

File: metrics/test/features/dashboard/domain/usecases/receive_build_metrics_updates_test.dart [hunk 4]
```diff
@@ -8,7 +10,7 @@ import 'package:metrics/util/date.dart';
 import 'package:test/test.dart';
 
 void main() {
-  group("Build metrics data loading", () {
+  group("ReceiveProjectMetricUpdates", () {
```
Comment (illia-romanenko) [inline metrics/test/features/dashboard/domain/usecases/receive_build_metrics_updates_test.dart]: Can we add more edge cases? Like all failed builds loaded and we need to find build for coverage?

File: metrics/test/features/dashboard/presentation/widgets/titled_text_test.dart [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+          tester.takeException(),
+          isAssertionError,
+        );
+      },
+    );
+
+    testWidgets(
+      "can't be created with null value",
+      (WidgetTester tester) async {
+        await tester.pumpWidget(const TitledTextTestbed(
```
Comment (illia-romanenko) [inline metrics/test/features/dashboard/presentation/widgets/titled_text_test.dart]: Should we provide `title`, in this case, to make sure that it's failing actually due to value null?

File: metrics/lib/features/dashboard/presentation/model/project_metrics_data.dart [hunk 6]
```diff
@@ -36,7 +34,6 @@ class ProjectMetricsData {
     this.numberOfBuilds,
```
Comment (illia-romanenko) [inline metrics/lib/features/dashboard/presentation/model/project_metrics_data.dart]: Maybe we should still have buildNumberMetrics but remove numberOfBuilds? Notice description:

/// [numberOfBuilds] is the number of builds the [buildNumberMetrics] are based on.

it still refers to it and I guess it will be more clear if we refer it as metric. What do you think?

File: metrics/lib/features/dashboard/domain/usecases/receive_project_metrics_updates.dart (hunk 2 above)
Comment (illia-romanenko) [reply metrics/lib/features/dashboard/domain/usecases/receive_project_metrics_updates.dart]: ?

File: metrics/test/features/dashboard/presentation/widgets/text_metric_test.dart [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+
+        final titleWidget = tester.widget<Text>(
+          find.text(TextMetricTestbed.defaultTitleText),
+        );
+        final valueWidget = tester.widget<Text>(
+          find.text(TextMetricTestbed.defaultValueText),
+        );
+
+        expect(titleWidget.style.color, titleStyle.color);
+        expect(valueWidget.style.color, valueStyle.color);
```
Comment (vladlaba) [inline metrics/test/features/dashboard/presentation/widgets/text_metric_test.dart]: Maybe it would be better to divide this into two test cases?

File: metrics/test_driver/app_test.dart [hunk 8]
```diff
@@ -35,15 +35,23 @@ void main() {
...
           await driver.waitFor(find.byType('SparklineGraph'));
         },
       );
 
+      test(
+        'Loads the build number metric and shows it with the title',
+        () async {
+          await driver.waitFor(find.text(DashboardStrings.builds));
+
+          await driver.waitFor(find.byType('TitledText'));
```
Comment (illia-romanenko) [inline metrics/test_driver/app_test.dart]: isn't it new type now? TextMetric?

File: metrics/test/features/dashboard/presentation/widgets/project_metrics_tile_test.dart [hunk 9]
```diff
@@ -74,12 +74,12 @@ void main() {
   );
 
   testWidgets(
-    'Contains SparklineGraph widgets with build metric',
+    'Contains total number of builds for last 7 days',
```
Comment (illia-romanenko) [inline metrics/test/features/dashboard/presentation/widgets/project_metrics_tile_test.dart]: how do we test here that it's for 7 days?

File: metrics/test/features/dashboard/domain/usecases/receive_project_metrics_updates_test.dart [hunk 10]
```diff
@@ -0,0 +1,242 @@
...
+
+    test('loads all fields in the build result metrics', () {
+      final buildResultMetrics = projectMetrics.buildResultMetrics;
+
+      final firstBuildResult = buildResultMetrics.buildResults.last;
+
+      expect(firstBuildResult.buildStatus, lastBuild.buildStatus);
+      expect(firstBuildResult.duration, lastBuild.duration);
+      expect(firstBuildResult.date, lastBuild.startedAt);
+      expect(firstBuildResult.url, lastBuild.url);
```
[the remark this answers is not in the document]
Comment (vova-beloded) [reply metrics/test/features/dashboard/domain/usecases/receive_project_metrics_updates_test.dart:77]: We can't extend `Equatable` and compare the `firstBuildResult` and the `lastBuild` because the `firstBuildResult` is the `BuildResult` and the `lastBuild` is the `Build`. As for predicate matcher, I think it is better to compare each field separately because in case this test fails we will be notified on which filed it failed.
